.github/workflows/publish.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish to NPM
+on:
+  release:
+    types: [created]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        id: install-pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+          run_install: false
+
+      - name: Setup Node
```

Comment:
I'd recommend checking out this example for pnpm dependency caching and installation: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data